cmd/cli/cli.go

```
@@ -336,6 +352,10 @@ func handleWrite(cfg *config.Config, write func() error) error {
 	)
 
 	if err == nil {
+		// Reset monitoring only if previously there was a setup error
```

@oo-sleeper I think we could move this block out of `if err == nil` block, but with additional checks.

What error status means here:

1) For err nil - success, all good.
2) For err non nil - I think it could mean two things:

* It's not setup error, but transient error. So still error, but not kind of error that triggers alerts, so we should reset monitoring.
* (not sure) It's still setup error, but we run out attempts! I think in this case we should return this error from `handleWrite` and crash an app. @oo-sleeper could you check if that's indeed the case?

---

> I think in this case we should return this error from handleWrite and crash an app

I'd recon this makes sense, but feels like a big change? @colmsnowplow

---

> So still error, but not kind of error that triggers alerts, so we should reset monitoring.

giving this one a think - why should we reset? data still cannot be sent, so heartbeat shouldn't be triggered, not alerts would be sent because it is then transient error which cannot be fixed by the user 🤔 

so code seem to be legit rn

---

> data still cannot be sent, so heartbeat shouldn't be triggered,

Heartbeat means "the app is functioning as expected, and doesn't require any user change to continue to process data". We can more easily define it as "there is not a setup error".

If data cannot be sent, and it's not a setup error, it is one of two scenarios:

- Transient error - it hasn't been successful, but we expect retries to resolve it. The app is functioning as expected, and can process data. Heartbeat should still be sent. IF retrying ISN'T successful, we will end up crashing the app and rebooting and infra alerting will come into play.
- Invalid error - it hasn't been successful, because the data is invalid. We sink it to the failure target and move on. The app is processing data as expected, and we should send heartbeat event.

We might change or refine these behaviours later, but this is what the rules are at the moment. :)

---

so I am hearing that if it is transient error, then we should not be reporting transient error after initial setup retries (where we did send an setup error as an alert) and monitoring still continues

in this case it is a simple change, adding it now 🤝